src/HistoryAPI.ts

```diff
@@ -110,21 +114,64 @@ function getPositions(
 
   debug(query)
 
+  if (format) format = format.toLowerCase()
+
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   v1Client.query(query).then((rows: any[]) => {
-    const resultData = rows.map((row) => {
-      return [row.time.toISOString(), [row.lon, row.lat]]
-    })
-
-    res.json({
-      context,
-      range: {
-        from: from.toString(),
-        to: to.toString(),
-      },
-      values: [{ path: 'navigation.position', method: 'first' }],
-      data: resultData,
-    })
+    if (format && format !== 'json') {
+      // output data in custom format
+      if (supportedFormats.includes(format)) {
+        if (format === 'gpx') {
+          let responseBody = `<?xml version="1.0" encoding="UTF-8" ?>
+<gpx xmlns="http://www.topografix.com/GPX/1/1" version="1.1" creator="signalk-to-influxdb2">
+<metadata><author>${context}</author></metadata>
+<trk>`
+          let inSegment = false;
+          rows.forEach((p) => {
+            if (p.lat != null && p.lon != null) {
+              if (!inSegment) {
+                responseBody += '\n<trseg>'
+                inSegment = true
+              }
+              responseBody += `<trkpt lat="${p.lat}" lon="${p.lon}"><time>${p.time.toISOString()}</time></trkpt>`
+            } else {
+              if (inSegment) {
+                responseBody += '</trseg>'
+                inSegment = false
+              }
+            }
+          })
+          if (inSegment) responseBody += '</trseg>'
+          responseBody += `
```

Comment:
very minor, but `'\n</trk>\n</gpx'` would look cleaner to me than the out of place indentations